qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,343 @@
+        optimizer (Optimizer | Minimizer): A classical optimizer to find the minimum energy. This
+            can either be a Qiskit :class:`.Optimizer` or a callable implementing the
+            :class:`.Minimizer` protocol.
+        gradient (BaseEstimatorGradient | None): An optional estimator gradient to be used with the
+            optimizer.
+        initial_point (Sequence[float] | None): An optional initial point (i.e. initial parameter
+            values) for the optimizer. The length of the initial point must match the number of
+            :attr:`ansatz` parameters. If ``None``, a random point will be generated within certain
+            parameter bounds. ``VQE`` will look to the ansatz for these bounds. If the ansatz does
+            not specify bounds, bounds of :math:`-2\pi`, :math:`2\pi` will be used.
+        callback (Callable[[int, np.ndarray, float, dict], None] | None): A callback that can access
+            the intermediate data at each optimization step. These data are: the evaluation count,
+            the optimizer parameters for the ansatz, the evaluated mean, and the metadata
+            dictionary.
+
+    References:
+        [1] Peruzzo et al, "A variational eigenvalue solver on a quantum processor"
+            `arXiv:1304.3061 https://arxiv.org/abs/1304.3061>`_
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        ansatz: QuantumCircuit,
+        optimizer: Optimizer | Minimizer,
+        *,
+        gradient: BaseEstimatorGradient | None = None,
+        initial_point: Sequence[float] | None = None,
+        callback: Callable[[int, np.ndarray, float, dict], None] | None = None,
+    ) -> None:
+        """
+        Args:
+            estimator: The estimator primitive to compute the expectation value of the
+                Hamiltonian operator.
+            ansatz: A parameterized quantum circuit to prepare the trial state.
+            optimizer: A classical optimizer to find the minimum energy. This
+                can either be a Qiskit :class:`.Optimizer` or a callable implementing the
+                :class:`.Minimizer` protocol.
+            gradient: An optional estimator gradient to be used with the optimizer.
+            initial_point: An optional initial point (i.e. initial parameter values) for the optimizer.
+                The length of the initial point must match the number of :attr:`ansatz` parameters.
+                If ``None``, a random point will be generated within certain parameter bounds.
+                ``VQE`` will look to the ansatz for these bounds. If the ansatz does not specify
+                bounds, bounds of :math:`-2\pi`, :math:`2\pi` will be used.
+            callback: A callback that can access the intermediate data at each optimization step.
+                These data are: the evaluation count, the optimizer parameters for the ansatz, the
+                evaluated mean, and the metadata dictionary.
```

Comment:
Maybe we should say `estimated value` instead of `evaluated mean` - since its the estimated observable value provided by the estimator whatever that is.